src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/RemoveAzureRmVpnClientIpsecParametersCommand.cs

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Commands.Network.VirtualNetworkGateway;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmVpnClientIpsecParameter", DefaultParameterSetName = "ByFactoryName", SupportsShouldProcess = true)]
+    public class RemoveAzureVpnClientIpsecParametersCommand : VirtualNetworkGatewayBaseCmdlet
+    {
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryName,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The virtual network gateway name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string VirtualNetworkGatewayName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryName,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryObject,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The virtual network gateaway object")]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualNetworkGateway InputObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceId,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Azure resource ID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Don't ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
```

Comment:
@Nilambari the `Force` parameter should only be used in cmdlets that require additional prompting outside of the typical `ShouldProcess` prompt. If this cmdlet does not prompt the user outside of that, then you should remove it

Reply:
Done.